storage/remote/queue_manager.go

```diff
@@ -207,6 +207,8 @@ type QueueManager struct {
 	samplesIn, samplesDropped, samplesOut, samplesOutDuration *ewmaRate
 	integralAccumulator                                       float64
 	startedAt                                                 time.Time
+	lastSendTimestampLock                                     sync.RWMutex
+	lastSendTimestamp                                         int64
```

Comment:
Note that if going the atomic route, the variable should be moved to the top of the struct to avoid breaking some builds: https://golang.org/pkg/sync/atomic/#pkg-note-BUG